### ISSUE TITLE
Remove support to parse other than `*.robot` files by default

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#3084]
Parsing other than `*.robot` files was deprecated in RF 3.1 for reasons explained in #2820 in RF 3.2 only `*.robot` files will be parsed by default. The support for HTML files will be removed for good by #3081 but files containing plain text data (incl. compatible TSV files and reStructuredText files with embedded data) can be used if `--extension` is used to enable them.

Things to consider:
- Should we try to look do reST files contain Robot Framework data and parse them automatically if they do? Would make using reST data easier but would complicate logic. Another possibility is somehow making global configuration easier.
- What to do with resource files using other extensions? They weren't deprecated by #2820. Should that be done now?

**Comment by yahman72** [comment]
My two cents worth to:
> What to do with resource files using other extensions? They weren't deprecated by #2820. Should that be done now?

I like the possibility of having different extension for the resource files. This way it's easy to identify the test suites (`*.robot`) from the rest. I see this quite a lot in scenarios where non-developers are involved in Test Case definition/review i.e. they work mostly with test cases (i.e. `*.robot` files) and the rest (resource files and libraries) is just black magic that makes things happen.
On the other hand `.txt` extension is also a bit misleading for the resource files. Maybe something new would be better, e.g. `.resource` 

In short: if we deprecate resource files using other extensions then add a new extension that is dedicated to resource files.

**Comment by pekkaklarck** [comment]
I agree `*.resource` extension is a good idea. That's why it was introduced in RF 3.1 by #2891. =)

**Comment by yahman72** [comment]
I though I remembered seeing it somewhere =)

**Comment by pekkaklarck** [comment]
This has now been implemented like this:

- When executing a single file, it is parsed regardless the extension. Allows running `robot tests.rst` as earlier and even supports `robot tests.whatever`.

- When executing a directory, only `*.robot` files are parsed by default. The `--extension` option needs to be used to parse other kind of files, including reST files.

At the moment resource file extension isn't validated at all but that still needs to change. Options include:

- Use same extensions as used for parsing and in addition that support also the explicit `*.resource` extension introduced by #2891. This would be the best solution but it may not be that easy to pass  `--extension` information to the resource file parsing code.

- Support only `*.resource` , `*.robot` and `*.r(e)st` files.

- Support `*.resource` , `*.robot`, `*.r(e)st`, `*.tsv` and `*.txt` files.

Once resource files are handled somehow, this issue can be considered done. Documenting changes in the User Guide is covered by #3293.

**Comment by pekkaklarck** [comment]
I think it's best to just support `*.resource` , `*.robot`, `*.r(e)st`, `*.tsv` and `*.txt` extensions with resource files. This is easy to implement and, because these are the extensions we supported earlier, it shouldn't cause backwards compatibility problems.
